### PR TITLE
Testenv machine

### PR DESCRIPTION
**Pull request by outscale-vbr**
Add functional test for machine controller

### CODE REVIEW

**Comment by outscale-vbr** [comment]
wait https://github.com/outscale-dev/cluster-api-provider-outscale/pull/38 is merged to rebase

**Comment by outscale-vbr** [comment]
(testenv pass )
https://github.com/outscale-vbr/cluster-api-provider-outscale/actions/runs/2868661153

**Comment by outscale-vbr** [comment]
(all test pass localy)

**Comment by outscale-vbr** [comment]
some part of webhook are generated:
https://book.kubebuilder.io/cronjob-tutorial/webhook-implementation.html

**Comment by outscale-vbr** [comment]
use cloud init script also (to have same bahavoir as cluster-api expected (bootstrap))
https://cluster-api.sigs.k8s.io/developer/architecture/controllers/machine.html
Machine.Spec.Bootstrap.DataSecretName

**Comment by outscale-vbr** [comment]
cluster-machine-simple will evolve in futur pr to have volume optional and use rootDisk.

File: cloud/scope/machine.go [hunk 1]
```diff
@@ -2,12 +2,12 @@ package scope
 
 import (
 	"context"
-
 	"errors"
 	"fmt"
 	osc "github.com/outscale/osc-sdk-go/v2"
 	corev1 "k8s.io/api/core/v1"
 	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+
```
Comment (outscale-hmi) [inline cloud/scope/machine.go:10]: why this blanc line ?

File: controllers/oscmachine_controller.go [hunk 2]
```diff
@@ -23,6 +23,10 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/source"
 	"time"
 
+	"sigs.k8s.io/cluster-api/util/predicates"
+
```
Comment (outscale-hmi) [inline controllers/oscmachine_controller.go:27]: two blanc lines  ?

File: controllers/oscmachine_controller.go [hunk 3]
```diff
@@ -326,7 +329,15 @@ func (r *OscMachineReconciler) reconcileDelete(ctx context.Context, machineScope
 
 // SetupWithManager sets up the controller with the Manager.
 func (r *OscMachineReconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manager, options controller.Options) error {
-	err := ctrl.NewControllerManagedBy(mgr).
+	clusterToObjectFunc, err := util.ClusterToObjectsMapper(r.Client, &infrastructurev1beta1.OscMachineList{}, mgr.GetScheme())
+
+	if err != nil {
+
```
Comment (outscale-hmi) [inline controllers/oscmachine_controller.go:335]: blanc line

File: controllers/oscmachine_vm_controller.go [hunk 4]
```diff
@@ -214,6 +213,7 @@ func checkVmFormatParameters(machineScope *scope.MachineScope, clusterScope *sco
 			return vmTagName, err
 		}
 	}
+
```
Comment (outscale-hmi) [inline controllers/oscmachine_vm_controller.go:216]: why this blanc line

File: controllers/oscmachine_controller.go [hunk 5]
```diff
@@ -326,7 +329,15 @@ func (r *OscMachineReconciler) reconcileDelete(ctx context.Context, machineScope
...
+		return errors.Errorf("failed to create mapper for Cluster to OscMachines: %+v", err)
+
+	}
+
+	err = ctrl.NewControllerManagedBy(mgr).
 		For(&infrastructurev1beta1.OscMachine{}).
 		Watches(
```
Comment (outscale-hmi) [inline controllers/oscmachine_controller.go:342]: the format here ?

File: controllers/oscmachine_vm_controller.go [hunk 6]
```diff
@@ -400,14 +400,15 @@ func reconcileVm(ctx context.Context, clusterScope *scope.ClusterScope, machineS
...
-				return reconcile.Result{}, fmt.Errorf("%s Can not create inbound securityGroupRule for OscCluster %s/%s", err, clusterScope.GetNamespace(), clusterScope.GetName())
+				return reconcile.Result{}, fmt.Errorf("%w Can not create inbound securityGroupRule for OscCluster %s/%s", err, clusterScope.GetNamespace(), clusterScope.GetName())
 			}
 		}
 
 		machineScope.Info("#### Get Vm ###", "vm", vm)
 		vmRef.ResourceMap[vmName] = vmID
 		vmSpec.ResourceId = vmID
 		machineScope.SetProviderID(vmID)
+
```
Comment (outscale-hmi) [inline controllers/oscmachine_vm_controller.go:411]: blanc line ?

**Review by outscale-hmi — COMMENTED** [review]
format issues but otherwise seems ok for me

File: controllers/oscmachine_controller.go (hunk 3 above)
Comment (outscale-vbr) [reply controllers/oscmachine_controller.go]: done

File: cloud/scope/machine.go (hunk 1 above)
Comment (outscale-vbr) [reply cloud/scope/machine.go]: done

**Review by jerome-jutteau — APPROVED** [review]

**Review by outscale-hmi — APPROVED** [review]
OK

**Pull request merged**